### DISCUSSION TITLE
How does Sioyek associate document with annotations?

### DISCUSSION THREAD

**Discussion by martinsifrar** [ahrm/sioyek#695] [General]
Annotations are stored in the `shared.db` file. On top of that, I'd like to know how Sioyek remembers what annotations belong to what file? What happens when documents are moved/renamed?

**Comment by ahrm** [comment]
We use the hash of the document so it still works when the document is moved/renamed (but not when it is modified).

**Reply by Michael-Gerlach** [reply]
Unfortunately, on my mac this does not work. When I move or rename a file, the hash of the file is conserved and still matches the hash in local.db, but the highlights and bookmarks to this file are lost to sioyek. They reappear when I move the file back to the location sioyek is looking for it. However, this means I can neither rename nor move a document.

**Reply by Michael-Gerlach** [reply]
Okay, it seems like Sioyek will recover a bookmarks and highlights eventually, sometimes I have to open it twice or wait a bit. It would be great to have the option to rename the file from within Sioyek, so links to bookmarks etc are kept working continuously.

**Reply by ahrm** [reply]
This is already fixed, in the development branch we don't need restart to reload notifications, we also have a `rename` command to rename the file from sioyek.
